org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -104,4 +124,93 @@ protected void initializeTypeProvider(XtextResourceSet resourceSet) {
 		resourceSet.setClasspathURIContext(getJavaProject(resourceSet));
 	}
 
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, CharSequence content) {
+		return createDslFile(fileName, fileExtensionProvider.getPrimaryFileExtension(), content);
+	}
+
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param fileExtension
+	 * 		The extension of the file.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, String fileExtension, CharSequence content) {
+		try {
+			return IResourcesSetupUtil.createFile(javaProject.getElementName(), fileName, fileExtension, content.toString());
```

Comment:
it would be useful to specify also the path of the file, e.g., into the "src" directory. Does this allow that? If so, maybe it should be specified in the documentation.

Reply:
To specity the worskpace relative path, the `IResourcesSetupUtil#createFile(String, String)` method can be used. I added this hint to the documentation.

Reply:
So, I don't understand the benefit of this new method... again, I'm probably missing something, but why not simply consider the path in the first place?

Reply:
This `createDslFile` method is basically a shorter way of calling the `IResourcesSetupUtil.createFile` method. The user can call this method if he/she wants to pass less parameter and does not want to bother with the exception handling. But if that does not fullfill his/her usecase, than he/she can call directly the `IResourcesSetupUtil.createFile` method instead.

Reply:
OK, but in any case, can the passed fileName consist of a path? I think that's important. I mean, it's important to be able to specify a path, otherwise that method is unusable for Xbase languages where files must be in source folders.

Reply:
@LorenzoBettini : I extended the javadoc as follows:

```
@param fileName
	 * 		The name of the file (without the file extension).
	 * 		It is also possible to use project relative path something like: <code>src/foo/bar/Test</code>
	 * 	 	To be able to specify the workspace relative path, use {@link IResourcesSetupUtil#createFile(String, String)} instead.
	 * 
```